goon/browserassets/js/browserOutput.js

```diff
@@ -1042,15 +1073,30 @@ $(function() {
 			count++;
 		}
 
+		opts.highlightRegexEnable = document.querySelector("#highlightRegexEnable").checked;
+		if (opts.highlightRegexEnable) {
+			// Check that all the patterns are valid regexes.
+			for(var i = 0; i < opts.highlightTerms.length; i++){
+				if (opts.highlightTerms[i]){
+					try {
+						new RegExp(opts.highlightTerms[i], "gmi");
+					} catch(e) {
+						var message = "Your highlight pattern '" + opts.highlightTerms[i] + "' is not a valid regex. Regex mode was automatically disabled.";
+						opts.highlightRegexEnable = false;
```

Comment:
The feature shouldnt literally disable itself when it encounters an error, this can also cause issue with the option enable being desyncronized from the actual value in cookies.

Reply:
Disabling the feature is the least bad behavior imo.
Feature being desynchronized from the cookies is possible, yes, but only realistic during the transition period. Reopening the highlights dialog would fix it (by disabling the regex), and you'd never be bothered with it again.

Reply:
opts is meant for the user's preferences, and generally shouldnt be touched to preserve the user's actual preferences

Reply:
using the `regexHasError` system here to do this without desyncing the opts object from the cookie might be a good thing to do
as well as intercepting the saving of the patterns to the cookie if they're invalid